src/core/execute.c

```diff
@@ -2412,12 +2413,14 @@ static int setup_exec_directory(
                                         goto fail;
                         }
 
-                        /* And link it up from the original place. Note that if a mount namespace is going to be
-                         * used, then this symlink remains on the host, and a new one for the child namespace will
-                         * be created later. */
-                        r = symlink_idempotent(pp, p, true);
-                        if (r < 0)
-                                goto fail;
+                        if (!context->directories[type].items[i].only_create) {
+                                /* And link it up from the original place. Note that if a mount namespace is going to be
+                                 * used, then this symlink remains on the host, and a new one for the child namespace will
+                                 * be created later. */
+                                r = symlink_idempotent(pp, p, true);
+                                if (r < 0)
+                                        goto fail;
+                        }
```

Comment:
I don't grok this change. `only_create` means we only do mkdir(), but not symlink(), did I get that right?

But what does that have to do with the issue this is supposed to fix? i can't see the relationship? that issue doesn't even use symlinks? what am I missing? can you explain?

Reply:
E.g. `StateDirectory=foo foo/bar`, then when processing `"foo/bar"`, `pp = "/var/lib/private/foo/bar"` and `p = "/var/lib/foo/bar"`. But there already exists a symlink `/var/lib/private/foo -> /var/lib/foo`. So, the inode of `pp` and `p` are the same, and we cannot create the symlink.